### ISSUE TITLE
Should the two reference implementations be API compatible?

### ISSUE THREAD

**Issue by arlolra** [coniks-sys/libmerkleprefixtree-go#25]
There a few known differences so far:

- Hash of STR sig instead of serialized STR
- Root is an interior node and serializes as the hash its branches
- Integer epochs
- Probably more ...

My vote is for yes, but I'm not sure I have a vote.

**Comment by masomel** [comment]
I totally agree. I've been really hoping for more community input, so your vote totally counts. I was going to wait to make these changes when I move a lot of the operations to a Java library, but I can make these changes now and move them along with everything else later.

**Comment by arlolra** [comment]
There's no rush.  Doing it at your leisure is fine.  We can continue to note differences as they come up here if it's helpful.

It'd be nice to eventually produce a shared test suite that we can run against both implementations.

**Comment by masomel** [comment]
> We can continue to note differences as they come up here if it's helpful.

Yeah, I think keeping track of the differences here will definitely be helpful.

> It'd be nice to eventually produce a shared test suite that we can run against both implementations.

Agreed. I don't know if this is necessary, but it might be helpful to keep the API versions in sync as well.
